src/handlers/feature_handlers.cpp

```diff
@@ -5201,6 +5201,25 @@ void vehCarModelFeatureHandler::DrawHeadlights(bool rotate)
     if (bothLightsBroken)
         return;
 
+    bool glowLoaded = false;
+    gfxTexture* headlightTexture;
+
+    //first time texture load
+    if (!glowLoaded) {
```

Comment:
`glowLoaded` and `headlightTexture` aren't static, so this condition will always be true (note I also wouldn't recommend making them static inside the function, because that would cause issues between races).

It would be much better to load/free them at the same time as the original `ltLight::GlowTexture` (inside `ltLight::ltLight` and `ltLight::ShutdownLights`).

Reply:
The condition should be always true to avoid loading the texture every time which means this should be loaded once. I don't think there's something wrong with this, the changes you wanna me to do can't be made because it's not easy to do as you think. If you check out `dgBangerInstance::DrawGlow`, you will find that it had been implemented in a similar way setting up the red glow texture to be used for banger glows.

Reply:
Always true, as in the if statement always executes. You are setting `glowLoaded = false;`, then immediately checking `!glowLoaded`, which will always evaluate to true.